src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -777,22 +830,38 @@ protected virtual bool TryLoadFromCache(JsonOperationContext context)
                 }
                 catch (HttpRequestException e) // server down, network down
                 {
-                    if (shouldRetry == false)
-                        throw;
+                    using (GetContext(out var requestContext))
+                    {
+                        if (e.InnerException is SocketException se && ShouldRemoveHttpClient(se))
+                        {
+                            if (requestContext.HttpClientRemoved == false)
+                            {
+                                RemoveHttpClient();
+                                requestContext.HttpClientRemoved = true;
+                            }
+                            else
+                            {
+                                requestContext.HttpClientRemoved = false;
```

Comment:
why are you setting this to the value it is already set to?

Reply:
I'm setting this from `true` to `false` because I've already tried that once and I now want `HandleServerDown` to behave correctly (return `false` if there is no other node to check and throw appropriate exception)